// Copyright 2016 The Cartographer Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

import "cartographer/mapping_3d/proto/projection_integrator_options.proto";

package cartographer.mapping_3d.proto;

message TSDFsOptions {
  // Resolution of the 'high_resolution' map in meters used for local SLAM and
  // loop closure.
  optional double high_resolution = 1;

  // Maximum range to filter the point cloud to before insertion into the
  // 'high_resolution' map.
  optional double high_resolution_max_range = 4;

  // Resolution of the 'low_resolution' version of the map in meters used for
  // local SLAM only.
  optional double low_resolution = 5;

  // Number of scans before adding a new submap. Each submap will get twice the
  // number of scans inserted: First for initialization without being matched
  // against, then while being matched.
  optional int32 num_range_data = 2;

  // Number of scans before adding a new submap. Each submap will get twice the
  // number of scans inserted: First for initialization without being matched
  // against, then while being matched.
  optional int32 chuck_size_x = 6;
  optional int32 chuck_size_y = 7;
  optional int32 chuck_size_z = 8;

  optional ProjectionIntegratorOptions projection_integrator_options = 3;
}
